yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -117,6 +119,25 @@ private[spark] class ApplicationMaster(
 
   private var delegationTokenRenewerOption: Option[AMDelegationTokenRenewer] = None
 
+  if (SystemUtils.IS_OS_UNIX) {
+    // Register signal handler for signal "TERM", "INT" and "HUP". For the cases where AM receive a
+    // signal and stop, from RM's aspect this application needs to be reattempted, rather than mark
+    // as success.
+    // Replace this signal handler with SignalLogger in AM side.
+    val signalHandler = new SignalHandler() {
+      override def handle(sig: Signal): Unit = {
+
+        logInfo(s"RECEIVED SIGNAL ${sig.getNumber}: SIG${sig.getName}")
```

Comment:
Yes, that also can be worked, it is just two implementation choices. I will change to that way.